src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -979,36 +980,36 @@ VideoDecoder::AVFrameStream VideoDecoder::decodeAVFrame(
     // We still haven't found the frame we're looking for. So let's read more
     // packets and send them to the decoder.
     ReferenceAVPacket packet(autoAVPacket);
-    ffmpegStatus = av_read_frame(formatContext_.get(), packet.get());
-    decodeStats_.numPacketsRead++;
+    bool foundPacketForStream = false;
```

Comment:
What happens in the loop below is barely changed. I only updated a `continue` into a `goto`.